src/component/project/Project/DeleteProject/DeleteProjectDialogue.tsx

```diff
@@ -0,0 +1,53 @@
+import { Dialogue } from '../../../common/Dialogue/Dialogue';
+import React from 'react';
+import { formatUnknownError } from '../../../../utils/formatUnknownError';
+import useProjectApi from '../../../../hooks/api/actions/useProjectApi/useProjectApi';
+import useProjects from '../../../../hooks/api/getters/useProjects/useProjects';
+import useToast from '../../../../hooks/useToast';
+
+interface IDeleteProjectDialogueProps {
+    project: string;
+    open: boolean;
+    onClose?: (e: React.SyntheticEvent, reason?: string) => void;
+    onFinish?: () => void;
+    onSuccess?: () => void;
+}
+
+export const DeleteProjectDialogue = ({
+    open,
+    onClose,
+    project,
+    onFinish,
+    onSuccess,
+}: IDeleteProjectDialogueProps) => {
+    const { deleteProject } = useProjectApi();
+    const { refetch: refetchProjectOverview } = useProjects();
+    const { setToastData, setToastApiError } = useToast();
+    return (
+        <Dialogue
+            open={open}
+            onClick={async (e: React.SyntheticEvent) => {
```

Comment:
Might be worth extracting this into a function / handler so it's more readable.

Reply:
Could a solution for both this and https://github.com/Unleash/unleash-frontend/pull/1207#discussion_r938826791 be to extract this as a `onDelete` prop that is provided by the parent components?

At that point we might not need a dedicated `DeleteProjectDialogue` component and instead just use a plain `Dialogue` 🤔 Something to think about.

Reply:
Moved it as separate function.

I dont think we should duplicate this on parent layer. Then we will have two almost identical methods on parent layer. The whole point of creating reusable component was to move this logic to where it belongs.